src/main/java/Collections/MyArrayList.java

```diff
@@ -32,13 +37,13 @@ private void increaseCapacity() {
     public boolean remove(T element) {
         for(int i = 0; i < array.length; i++){
             if(array[i] == element){
```

Comment:
This will always compare objects by their reference in the memory. Thus, it will never work for generic types meant to be boxed.